experimental/traffic-portal/src/app/shared/import-json-txt/import-json-txt.component.html

```diff
@@ -0,0 +1,38 @@
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<h2 mat-dialog-title>{{data.title}}</h2>
+<mat-dialog-content id="import-content" [ngClass]="{'active':dragOn}">
+	<input type='file' id="imageUpload" accept="application/json,text/plain" #fileInput (change)="uploadFile($event)"/>
+	<button type="button" class="dropzone" (click)="fileInput.click()">
+		<div class="text-wrapper">
+			<div class="label">
+				Drop your file here!<br>
+				<small class="hint">{{mimeAlertMsg}}</small>
+			</div>
+		</div>
+	</button>
+
+	<ul id="file-name">
+		<li *ngIf="fileData">{{fileData}}</li>
+	</ul>
+
+	<div id="json-txt" *ngIf="inputTxt">
+		<pre>{{inputTxt | json}}</pre>
+	</div>
+</mat-dialog-content>
+<mat-dialog-actions align="end">
+	<button mat-raised-button type="button" mat-dialog-close color="warn">Cancel</button>
+	<button mat-raised-button type="button" [mat-dialog-close]="true" (click)="submit()" [disabled]="!fileData">Submit</button>
```

Comment:
the `mat-dialog-close` directive allows you to bind to a value for the dialog's result, which you're doing here. But it doesn't do anything. Your dialog never closes with a `true` value because the click handler closes the dialog before that happens, returning the string content of the uploaded file instead. So if you want to use `mat-dialog-close` instead of a form submittal, then there's no reason to also have the click handler.